support/SubmitScripts/OceanClang.sh

```diff
@@ -76,13 +76,28 @@ for node in $(echo $SLURM_NODELIST | scontrol show hostnames); do
   echo "host ${node}" >> nodelist.$SLURM_JOBID
 done
 
-# The 19 is there because Charm++ uses one thread per node for communication
-# Here, -np should take the number of nodes (must be the same as --nodes
+# The (SLURM_NTASKS_PER_NODE - 1) is there because Charm++ uses one thread per
+# node for communication
+# Here, ++np should take the number of nodes (must be the same as --nodes
 # in the #SBATCH options above).
 WORKER_THREADS_PER_NODE=$((SLURM_NTASKS_PER_NODE - 1))
 WORKER_THREADS=$((SLURM_NPROCS - SLURM_NNODES))
 SPECTRE_COMMAND="${SPECTRE_EXECUTABLE} ++np ${SLURM_NNODES} \
 ++p ${WORKER_THREADS} ++ppn ${WORKER_THREADS_PER_NODE} \
 ++nodelist nodelist.${SLURM_JOBID}"
 
-charmrun ${SPECTRE_COMMAND} --input-file ${SPECTRE_INPUT_FILE}
+
+# When invoking through `charmrun`, charm will initiate remote sessions which
+# will wipe out environment settings unless it is forced to re-initialize the
+# spectre environment between the start of the remote session and starting the
+# spectre executable
+echo "#!/bin/sh
+source /home/moxon/spectre/support/Environments/ocean_clang.sh
```

Comment:
Can this path be drawn from the variables at the top of the file instead of hard coding here?